src/Features/LanguageServer/Protocol/Handler/RequestContext.cs

```diff
@@ -56,6 +56,11 @@ namespace Microsoft.CodeAnalysis.LanguageServer.Handler
         /// </summary>
         public readonly Document? Document;
 
+        /// <summary>
+        /// The additional document that the request is for, if applicable. This comes from the <see cref="TextDocumentIdentifier"/> returned from the handler itself via a call to <see cref="IRequestHandler{RequestType, ResponseType}.GetTextDocumentIdentifier(RequestType)"/>.
+        /// </summary>
+        public readonly TextDocument? AdditionalDocument;
```

Comment:
At some point we might want to have the handler define a RequestContext<T> that tells us whether the handler is looking for a regular document, additional document, analyzer config document, etc.  But not blocking this PR.